src/lib.rs

```diff
@@ -41,1346 +53,988 @@ use std::{
     /// for the polynomial commitment scheme.
-    fn setup(max_degree: usize) -> Result<Self::UniversalParams, Self::Error>;
+    fn setup(max_degree: usize) -> Result<Self::Parameters, Self::Error>;
 
     /// Constructs public parameters when given as input the maximum degree `degree`
     /// for the polynomial commitment scheme from given seed
-    fn setup_from_seed(
-        max_degree: usize,
-        seed: &[u8],
-    ) -> Result<Self::UniversalParams, Self::Error>;
-
-    /// Specializes the public parameters for polynomials up to the given `supported_degree`
-    /// and for enforcing degree bounds in the range `1..=supported_degree`.
-    fn trim(
-        pp: &Self::UniversalParams,
-        supported_degree: usize,
-    ) -> Result<(Self::CommitterKey, Self::VerifierKey), Self::Error>;
-
-    /// Outputs a commitments to `polynomials`. If `polynomials[i].is_hiding()`,
-    /// then the `i`-th commitment is hiding up to `polynomials.hiding_bound()` queries.
-    /// `rng` should not be `None` if `polynomials[i].is_hiding() == true` for any `i`.
-    ///
-    /// If for some `i`, `polynomials[i].is_hiding() == false`, then the
-    /// corresponding randomness is `Self::Randomness::empty()`.
+    fn setup_from_seed(max_degree: usize, seed: &[u8]) -> Result<Self::Parameters, Self::Error>;
+
+    /// Computes the commitment of a single polynomial
+    fn commit(
+        ck: &Self::CommitterKey,
+        polynomial: &Polynomial<G::ScalarField>,
+        // Set `is_hiding = true` for hiding commitments
+        is_hiding: bool,
+        // rng for hiding randomness
+        rng: Option<&mut dyn RngCore>,
+    ) -> Result<(Self::Commitment, Self::Randomness), Self::Error>;
+
+    /// Single-point multi-poly open, for proving an evaluation claim `p(x) = y`
+    /// at a query point `x`.
+    /// Note: It is assumed that the state of the Fiat-Shamir random number generator `fs_rng`
+    /// is already bound to the evaluation claim, i.e. the commitment of `polynomial`,
+    /// the point `x`, and the value `y`.
     ///
-    /// If for some `i`, `polynomials[i].degree_bound().is_some()`, then that
-    /// polynomial will have the corresponding degree bound enforced.
-    fn commit<'a>(
+    // TODO: As this is the interface for a general homomorphic scheme, we should overthink
+    // if it makes sense to pass also the commitment of `polynomial`, even if not needed
+    // in the case of the dlog scheme.
+    fn open(
+        ck: &Self::CommitterKey,
+        polynomial: Polynomial<G::ScalarField>,
+        point: G::ScalarField,
+        // Set `is_hiding = true` for zk of the evaluation proof.
+        // TODO: let us rename it to `zk`
+        is_hiding: bool,
+        hiding_randomness: Self::Randomness,
+        fs_rng: &mut Self::RandomOracle,
+        // The optional rng for additional internal randomness of open()
+        rng: Option<&mut dyn RngCore>,
+    ) -> Result<Self::Proof, Self::Error>;
```

Comment:
I was thinking again about our interface for the single-point single-poly open. Even if the inner product argument does not need to know the commitment of the polynomial, it seems to me artificial not to provide it. All the more expressive prover/verifier functions require the commitments, so let us demand it for `fn open()` too.

Reply:
Did not see that we already mentioned this in the inline comment above.